cli_tools_tests/e2e/variables.go

```diff
@@ -0,0 +1,53 @@
+//  Copyright 2020 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package e2e
+
+const (
+	projectIDWithoutDefaultServiceAccountFlag                      = "project_id_without_default_service_account"
+	projectIDWithoutDefaultServiceAccountPermissionFlag            = "project_id_without_default_service_account_permission"
+	computeServiceAccountWithoutDefaultServiceAccountFlag          = "compute_service_account_without_default_service_account"
+	computeServiceAccountWithoutDefaultServiceAccountPermissionFlag = "compute_service_account_without_default_service_account_permission"
+)
+
+// Additional variables used for e2e testing
+var (
+	ProjectIDWithoutDefaultServiceAccount, ProjectIDWithoutDefaultServiceAccountPermission,
+	ComputeServiceAccountWithoutDefaultServiceAccount, ComputeServiceAccountWithoutDefaultServiceAccountPermission string
+)
+
+// GetServiceAccountTestVariables extract extra test variables related to service account from input variable map.
+func GetServiceAccountTestVariables(argMap map[string]string) bool {
```

Comment:
Do you foresee the test suite binary being invoked with different accounts and projects? If not, thoughts on putting the accounts directly in the file?

Reply:
I prefer to keep the flexibility by config it inconfig.yaml, just like what we did for pool-001, even I don't plan to change it actively. Maybe someday the test project is not available anymore, or some behaviors arelimited by the org policy.